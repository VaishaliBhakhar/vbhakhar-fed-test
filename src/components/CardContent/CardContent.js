import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CardBodyContainer = styled.div`
	display: flex;
`;

const CardBodyBox = styled.div`
	width: 50%;
	@media only screen and (max-width: 767px) and (min-width: 400px) {
		width: 100%;
	}
`;

const CardBodyContent = styled.div`
	padding: 2px;
	display: flex;
	word-break: break-all;
`;

const CardBodyBottomContent = styled.div`
	width: 100%;
	margin-top: 5%;
	display: flex;
	@media (max-width: 767px) {
		display: block;
	}
`;

const Icon = styled.i`
	line-height: 2.5;
	color: #fcc02c;
`;

export default function CardContent(props) {
	CardContent.propTypes = {
		user: PropTypes.shape([]).isRequired,
		birthDate: PropTypes.string.isRequired
	};
	const { user, birthDate } = props;
	return (
		<>
			<CardBodyContainer>
				<CardBodyBox>
					<CardBodyContent>Gender: {user.gender}</CardBodyContent>
					<CardBodyContent>Birthday: {birthDate}</CardBodyContent>
					<CardBodyContent>Age: {user.dob.age}</CardBodyContent>
				</CardBodyBox>
				<CardBodyBox>
					<CardBodyContent>
						Address: {user.location.street.number},{user.location.street.name}
					</CardBodyContent>
					<CardBodyContent>Pincode: {user.location.postcode}</CardBodyContent>
					<CardBodyContent>State: {user.location.state}</CardBodyContent>
				</CardBodyBox>
			</CardBodyContainer>
			<CardBodyBottomContent>
				<CardBodyBox>
					<CardBodyContent>
						<Icon className="fa fa-envelope mr-5 text-info" aria-hidden="true" />
						<CardBodyContent>{user.email}</CardBodyContent>
					</CardBodyContent>
				</CardBodyBox>
				<CardBodyBox>
					<CardBodyContent>
						<Icon className="fa fa-phone-square mr-5 text-success" aria-hidden="true" />
						<CardBodyContent>{user.phone}</CardBodyContent>
					</CardBodyContent>
				</CardBodyBox>
			</CardBodyBottomContent>
		</>
	);
}
