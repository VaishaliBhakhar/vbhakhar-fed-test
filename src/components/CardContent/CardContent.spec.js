import React from 'react';
import { render } from '@testing-library/react';

import CardContent from './CardContent';

describe('CardContent', () => {
	it('matches screenshot', () => {
		const user = {
			gender: 'male',
			name: { title: 'Mr', first: 'Dwayne', last: 'Carr' },
			location: {
				street: { number: 5997, name: 'New Street' },
				city: 'Kinsale',
				state: 'Carlow',
				country: 'Ireland',
				postcode: 47945,
				coordinates: { latitude: '20.0798', longitude: '-72.6822' },
				timezone: { offset: '+4:30', description: 'Kabul' }
			},
			email: 'dwayne.carr@example.com',
			login: {
				uuid: '42a97806-e8ff-4de0-a264-ab745dfd34fa',
				username: 'happyfish704',
				password: 'bowser',
				salt: '5F09Z6d6',
				md5: '035c246faf6e3252139552495a4efc14',
				sha1: '25b5ccfcf8c9615d77b20ed6dae6cc80551f026e',
				sha256: '1fc39344a560336390cf20118724e0363e9324bf3e8a50ddc1049fb78f1a090d'
			},
			dob: { date: '1983-02-16T11:42:09.700Z', age: 38 },
			registered: { date: '2010-01-11T11:33:23.053Z', age: 11 },
			phone: '031-898-2396',
			cell: '081-649-8810',
			id: { name: 'PPS', value: '6858462T' },
			picture: {
				large: 'https://randomuser.me/api/portraits/men/28.jpg',
				medium: 'https://randomuser.me/api/portraits/med/men/28.jpg',
				thumbnail: 'https://randomuser.me/api/portraits/thumb/men/28.jpg'
			}
		};
		const birthDate = user.dob.date;
		const { asFragment } = render(<CardContent user={user} birthDate={birthDate} />);

		expect(asFragment()).toMatchSnapshot();
	});
});
