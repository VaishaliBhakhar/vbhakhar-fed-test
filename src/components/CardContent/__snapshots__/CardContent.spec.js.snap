// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent matches screenshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ bUzZtt"
  >
    <div
      class="sc-gsTCUz kSXPkl"
    >
      <div
        class="sc-dlfnbm cDazqa"
      >
        Gender: male
      </div>
      <div
        class="sc-dlfnbm cDazqa"
      >
        Birthday: 1983-02-16T11:42:09.700Z
      </div>
      <div
        class="sc-dlfnbm cDazqa"
      >
        Age: 38
      </div>
    </div>
    <div
      class="sc-gsTCUz kSXPkl"
    >
      <div
        class="sc-dlfnbm cDazqa"
      >
        Address: 5997,New Street
      </div>
      <div
        class="sc-dlfnbm cDazqa"
      >
        Pincode: 47945
      </div>
      <div
        class="sc-dlfnbm cDazqa"
      >
        State: Carlow
      </div>
    </div>
  </div>
  <div
    class="sc-hKgILt dzCBKu"
  >
    <div
      class="sc-gsTCUz kSXPkl"
    >
      <div
        class="sc-dlfnbm cDazqa"
      >
        <i
          aria-hidden="true"
          class="sc-eCssSg fYIEdU fa fa-envelope mr-5 text-info"
        />
        <div
          class="sc-dlfnbm cDazqa"
        >
          dwayne.carr@example.com
        </div>
      </div>
    </div>
    <div
      class="sc-gsTCUz kSXPkl"
    >
      <div
        class="sc-dlfnbm cDazqa"
      >
        <i
          aria-hidden="true"
          class="sc-eCssSg fYIEdU fa fa-phone-square mr-5 text-success"
        />
        <div
          class="sc-dlfnbm cDazqa"
        >
          031-898-2396
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
